cmd/k8s-operator/exec.go

```diff
@@ -0,0 +1,979 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !plan9
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"compress/zlib"
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/netip"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"tailscale.com/client/tailscale/apitype"
+	tskube "tailscale.com/kube"
+	"tailscale.com/ssh/tailssh"
+	"tailscale.com/tailcfg"
+	"tailscale.com/tsnet"
+	"tailscale.com/tstime"
+	"tailscale.com/util/multierr"
+)
+
+// This file contains functionality related to forwarding 'kubectl exec' over
```

Comment:
There's a number of different types defined in this file, most with their own methods. As it's quite a big file as well, it might make code navigation a bit easier to split this into (roughly) a file for each type, and/or maybe also a sub-package for encapsulating SPDY-related stuff?

Reply:
I've split it into separate files. I'd probably rather not create a separate package yet because the websockets functionality will change how this is structured- but also not strongly opinionated on this.